pkg/reconciler/pipelinerun/controller.go

```diff
@@ -64,6 +74,10 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 				logger.Error(err)
 			}
 			psSigner.Backends = backends
+
+			if err := storage.WatchBackends(ctx, watcherStop, psSigner.Backends, cfg); err != nil {
```

Comment:
yes, the first time the controller starts, push to stop will fail as there are no previous goroutines, however that will not block because of `select ... case`

however note that `watcherStop` is initialized outside the `cfgStore := config.NewConfigStore() ...` call which means that each time the configmap is updated, it kills the previous watcher and then starts the new one.

https://go.dev/play/p/SET7r1uYBlf